src/main/java/se/liu/ida/hefquin/engine/queryproc/impl/optimizer/DynamicProgramming/DPBasedOptimizerHelper.java

```diff
@@ -0,0 +1,71 @@
+package se.liu.ida.hefquin.engine.queryproc.impl.optimizer.DynamicProgramming;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+public class DPBasedOptimizerHelper {
```

Comment:
I don't think we need an extra class for that. Simply move these two helper methods into `DPBasedJoinPlanOptimizer`.